pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
+  pywbme loggers is defined before the connection is configured and the
+  logger is attached to :class:`~pywbem.WBEMConnection` object with one of the
+  :class:`~pywbem.WBEMConnection` methods.
+  Simply defining the loggers through standard python logger hander methods
+  does not activate these loggers since the logger is not attached to a
+  :class:`~pywbem.WBEMConnection` method
 
+* The logger is activated by linking it to a :class:`~pywbem.WBEMConnection`
+  object using the `connection` parameter of the configure... method
 
-# pylint: disable=no-init
-class PywbemLoggers(six.with_metaclass(MetaPywbemLoggers)):
-    """
-    **Experimental:** *New in pywbem 0.11 as experimental.*
+* The pywbem loggers may be configured through :class:`~pywbem.WBEMConnection`
+  methods.
+  These methods configure and optionally activate each of the loggers:
 
-    Container for pywbem logger information when loggers are created. This
-    class is a singleton, there is only one set of data for a pywbem
-    instantiation.  Its goal is to defined named loggers from data input and
-    too record information about these named loggers in a dictionary for
-    use by the log functions
+  * :meth:`~pywbem.WBEMConnection.configure_api_logger`
 
-    This is defined with only class level object and methods as an
-    easy way to create a singleton. However, at least in python 2.6, you
-    cannot make some magic methods work (ex.__getItem__)
+  * :meth:`~pywbem.WBEMConnection.configure_http_logger`
 
-    There are two constructors for loggers:
+  * :meth:`~pywbem.WBEMConnection.configure_configure_all_loggers`
 
-    * :meth:`~pywbem._logging.PywbemLoggers.create_loggers` - Creates one or
-      more logger entries in this class and also in the python logging class
-      from an input string with the format defined in the create_loggers
-      method below.
+In addition there is a method
+:meth:`~pywbem._logging.define_loggers_from_string` which allows setting
+up the loggers from a single formatted string that is most useful
+in defining the loggers from a command line tool such as pywbemcli.
 
-    * :meth:`~pywbem._logging.PywbemLoggers.create_logger` - Creates a single
-      logger from the separate pywbem logging parameters supplied with the
-      method.
-    """
+The following examples define a different logger configuration options:
 
-    @classmethod
-    def create_loggers(cls, input_str, log_filename=None):
-        # pylint: disable=line-too-long
-        """
-        Create the pywbem loggers defined by the input string in the following
-        syntax and place the logger definitions in the class level dictionary
-        in this class.
+Configure a single pywbem connection with standard Python logger methods::
 
-        Parameters:
+    import logging
+    logging.basicConfig(filename='example.log',level=logging.DEBUG)
+    conn = WBEMConnection(...)
+    conn.configure_all_loggers(detail_level='all', connection=conn)
```

Comment:
As a class method, it should be called against the class: `WBEMConnection.configure...`

Reply:
Done